### PR TITLE
feat: faq migrations and models

### PR DESCRIPTION
**Pull request by Matistan**
sorry for being late, but I finally made migrations and models for FAQ section

### CODE REVIEW

File: app/models/guide_article_author.ts
Comment (mini-bomba) [inline app/models/guide_article_author.ts:1]: I don't think you don't need to create models for pivot tables

File: app/models/guide_author.ts
Comment (mini-bomba) [inline app/models/guide_author.ts:1]: This model is missing updatedAt/createdAt timestamps. Is this intentional?

File: database/migrations/1734288675568_create_guide_authors_table.ts
Comment (mini-bomba) [inline database/migrations/1734288675568_create_guide_authors_table.ts:1]: This table is missing updatedAt/createdAt timestamps. Is this intentional?

File: database/migrations/1734288366467_create_guide_articles_table.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  async up() {
+    this.schema.createTable(this.tableName, (table) => {
+      table.increments("id").notNullable();
+      table.text("title").notNullable();
+      table.text("short_desc").notNullable();
+      table.text("description").notNullable();
+      table.text("image_path").notNullable();
+
+      table.timestamp("created_at");
+      table.timestamp("updated_at");
```
Comment (mini-bomba) [inline database/migrations/1734288366467_create_guide_articles_table.ts:15]: timestamp columns should be made not nullable
```suggestion
      table.timestamp("created_at").notNullable();
      table.timestamp("updated_at").notNullable();
```

File: database/migrations/1734288691900_create_guide_questions_table.ts
Comment (mini-bomba) [inline database/migrations/1734288691900_create_guide_questions_table.ts:1]: This table is missing updatedAt/createdAt timestamps. Is this intentional?

File: app/models/guide_question.ts
Comment (mini-bomba) [inline app/models/guide_question.ts:1]: This model is missing updatedAt/createdAt timestamps. Is this intentional?

File: app/models/guide_author.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  declare name: string;
+
+  @manyToMany(() => GuideArticle, {
+    pivotTable: "guide_article_authors",
+    pivotColumns: ["role"],
+    pivotForeignKey: "author_id",
+    pivotRelatedForeignKey: "article_id",
+    relatedKey: "id",
+  })
+  declare guideArticles: relations.ManyToMany<typeof GuideArticle>;
```
Comment (mini-bomba) [inline app/models/guide_author.ts:20]: might want to enable `pivotTimestamps` on this relation
```suggestion
  @manyToMany(() => GuideAuthor, {
    pivotTable: "guide_article_authors",
    pivotColumns: ["role"],
    pivotForeignKey: "article_id",
    pivotRelatedForeignKey: "author_id",
    relatedKey: "id",
    pivotTimestamps: true,
  })
  declare guideAuthors: relations.ManyToMany<typeof GuideAuthor>;
```

File: database/migrations/1734288708732_create_guide_article_authors_table.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      table
+        .enum("role", this.guideAuthorRoles, {
+          useNative: true,
+          enumName: "guide_author_role",
+          existingType: false,
+        })
+        .notNullable();
+
+      table.timestamp("created_at");
+      table.timestamp("updated_at");
```
Comment (mini-bomba) [inline database/migrations/1734288708732_create_guide_article_authors_table.ts:29]: timestamp columns should be made not nullable
```suggestion
      table.timestamp("created_at").notNullable();
      table.timestamp("updated_at").notNullable();
```

File: app/models/guide_article.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  declare updatedAt: DateTime;
+
+  @manyToMany(() => GuideAuthor, {
+    pivotTable: "guide_article_authors",
+    pivotColumns: ["role"],
+    pivotForeignKey: "article_id",
+    pivotRelatedForeignKey: "author_id",
+    relatedKey: "id",
+  })
+  declare guideAuthors: relations.ManyToMany<typeof GuideAuthor>;
```
Comment (mini-bomba) [inline app/models/guide_article.ts:38]: might want to enable `pivotTimestamps` on this relation
```suggestion
  @manyToMany(() => GuideAuthor, {
    pivotTable: "guide_article_authors",
    pivotColumns: ["role"],
    pivotForeignKey: "article_id",
    pivotRelatedForeignKey: "author_id",
    relatedKey: "id",
    pivotTimestamps: true,
  })
  declare guideAuthors: relations.ManyToMany<typeof GuideAuthor>;
```

**Review by mini-bomba — COMMENTED** [review]
initial review; minor issues

**Comment by Matistan** [comment]
faq tables
> This model is missing updatedAt/createdAt timestamps. Is this intentional?

Yeah, I just added it in the GuideArticles table, as it is in the schema. I can add them everywhere else if you want

> I don't think you don't need to create models for pivot tables

You mean that there is no need for guide_article_author.ts model?

**Comment by Matistan** [comment]
ok, after playing with editing names of already pushed commit's (forgot the convension from the docs) everything should be fine now

**Review by czaja307 — APPROVED** [review]
good job 🔥

**Pull request merged**